### PR TITLE
Create user service and default marker reporter is user 

### PR DESCRIPTION
**Pull request by NoyFlat**
Created user service so different components can get the current user data.
User service is updated by the authentication component.
Info window component gets the current user and displays the user name as the default reporter in marker.

### CODE REVIEW

File: capstone-project/src/main/angular-webapp/src/app/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { SocialUser } from "angularx-social-login";
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
```
Comment (NaamaHolzer) [inline capstone-project/src/main/angular-webapp/src/app/user.service.ts:7]: Forgot to document the class and the its method

File: capstone-project/src/main/angular-webapp/src/app/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export class UserService {
```
Comment (NaamaHolzer) [inline capstone-project/src/main/angular-webapp/src/app/user.service.ts:8]: Forgot to document the class and the its method

**Review by NaamaHolzer — APPROVED** [review]

File: capstone-project/src/main/angular-webapp/src/app/authentication/authentication.component.ts [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
   }
 
+  // Sign out user and reload page
   signOut(): void {
     this.authService.signOut();
+    window.location.reload();
+  }
+
+  // Return the current user
+  getUser(){
```
Comment (orarch) [inline capstone-project/src/main/angular-webapp/src/app/authentication/authentication.component.ts:33]: Try:

get user(): SocialUser {
  return this.userService.currentUser;
}

I think then you can access user in HTML as before (just with 'user' and not 'user()')

File: capstone-project/src/main/angular-webapp/src/app/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  // Holds the current user, updated in authentication component
+  currentUser;
```
Comment (orarch) [inline capstone-project/src/main/angular-webapp/src/app/user.service.ts:9]: Please make the field private and add a setter and getter.

**Review by orarch — APPROVED** [review]

File: capstone-project/src/main/angular-webapp/src/app/authentication/authentication.component.ts (hunk 3 above)
Comment (NoyFlat) [reply capstone-project/src/main/angular-webapp/src/app/authentication/authentication.component.ts:33]: Cool! Thanks

**Pull request merged**